### PR TITLE
feat: add additional log levels

### PR DESCRIPTION
**Pull request by joshka**
This adds `OffLevel`, `DebugLevel`, and `TraceLevel` to the list of log
levels that can be set as the default log level for the logger.

A new example, `log_level.rs`, demonstrates how to set the default log
level for Verbosity to something other than the default. It can be run
with `cargo run --example=log_level [default level] [flags]`. E.g.,:
`cargo run --example=log_level off -vvvv` will set the default log level
to `OffLevel` and then apply the verbosity flags to set the log level to
`Debug`.

Fixes: https://github.com/clap-rs/clap-verbosity-flag/issues/122

### CODE REVIEW

File: examples/log_level.rs
Comment (epage) [inline examples/log_level.rs:1]: This should at least be its own commit, maybe its own PR

File: src/lib.rs [hunk 1]
```diff
@@ -190,7 +192,6 @@ pub trait LogLevel {
 }
 
 /// Default to [`log::Level::Error`]
-#[allow(clippy::exhaustive_structs)]
```
Comment (epage) [inline src/lib.rs:193]: Why is this needed?

File: src/lib.rs [hunk 2]
```diff
@@ -222,6 +221,37 @@ impl LogLevel for InfoLevel {
...
+pub struct TraceLevel;
+
+impl LogLevel for TraceLevel {
+    fn default() -> Option<Level> {
+        Some(Level::Trace)
+    }
+}
+
+/// Default to no logging (i.e. `None` or [`log::LevelFilter::Off`])
+#[allow(clippy::exhaustive_structs)]
```
Comment (epage) [inline src/lib.rs:245]: Why is this on this struct but none of the others?

Comment (joshka) [reply src/lib.rs:245]: accidental copy paste issue - will remove.

File: src/lib.rs (hunk 1 above)
Comment (joshka) [reply src/lib.rs:193]: It's not, which is why it's removed.

File: examples/log_level.rs
Comment (joshka) [reply examples/log_level.rs:1]: 28a2888...31517ef

File: src/lib.rs (hunk 2 above)
Comment (joshka) [reply src/lib.rs]: 28a2888...31517ef

File: examples/log_level.rs [hunk 3]
```diff
@@ -0,0 +1,68 @@
+//! Demonstrates how to set the default log level for the logger to something other than the default
+//! (ErrorLevel). This is done with multiple subcommands, each with their own verbosity level.
```
[the remark this answers is not in the document]
Comment (joshka) [reply examples/log_level.rs]: 31517ef...7f9b94c

File: examples/log_level.rs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            Command::Info { verbose } => verbose.log_level_filter(),
+            Command::Debug { verbose } => verbose.log_level_filter(),
+            Command::Trace { verbose } => verbose.log_level_filter(),
+        }
+    }
+}
+
+fn main() {
+    let cli = Cli::parse();
+    dbg!(&cli.command);
```
[the remark this answers is not in the document]
Comment (joshka) [reply examples/log_level.rs]: 31517ef...7f9b94c

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -104,7 +104,7 @@ jobs:
     - name: Install Rust
       uses: dtolnay/rust-toolchain@stable
       with:
-        toolchain: "1.81.0"  # STABLE
+        toolchain: "1.82.0"  # STABLE
```
Comment (epage) [inline .github/workflows/ci.yml:107]: How did f1c663cba56e3d9a067b08725262703c78f79c05 become apart of this PR?

File: src/lib.rs (hunk 1 above)
Comment (epage) [reply src/lib.rs:193]: That is independent of adding log levels and should be its own commit

File: .github/workflows/ci.yml (hunk 5 above)
Comment (joshka) [reply .github/workflows/ci.yml:107]: renovate bot did it - it's not something I'm familiar with. Something enabled on this repo that automatically makes changes to PRs?

File: src/lib.rs (hunk 1 above)
Comment (joshka) [reply src/lib.rs:193]: e628191

File: .github/workflows/ci.yml (hunk 5 above)
Comment (joshka) [reply .github/workflows/ci.yml]: I'm unsure how this ended up being added to the PR, but I've rebased on master and it is no longer there.

**Comment by epage** [comment]
Thanks!
